src/todo.js

```diff
@@ -0,0 +1,149 @@
+// Array to store the tasks
+let tasks = [];
+
+// Function to update task indexes
+function updateIndexes() {
+  tasks.forEach((task, index) => {
+    task.index = index + 1;
+  });
+}
+
+// Function to save tasks to local storage
+export function saveTasksToLocalStorage() {
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+}
+
+// Function to load tasks from local storage
+export function loadTasksFromLocalStorage() {
+  const savedTasks = localStorage.getItem('tasks');
+  if (savedTasks) {
+    tasks = JSON.parse(savedTasks);
+  }
+}
+
+// Function to delete a task
+export function deleteTask(todoItem) {
+  /* eslint-disable */
+  const taskIndex = tasks.findIndex((task) => task.index === parseInt(todoItem.dataset.index));
+  if (taskIndex !== -1) {
+    tasks.splice(taskIndex, 1);
+    updateIndexes(); // Update task indexes after deleting a task
+    /* eslint-disable */
+    saveTasksToLocalStorage(); // Save updated tasks to local storage
+    renderTodoList(); // Render the updated todo list
+  }
+}
+// Function to edit a task
+export function editTask(todoItem) {
+  const taskElement = todoItem.querySelector('.task');
+  const taskText = taskElement.textContent;
+  taskElement.innerHTML = '';
+  const editInput = document.createElement('input');
+  editInput.type = 'text';
+  editInput.value = taskText;
+  editInput.classList.add('edit-input');
+  taskElement.appendChild(editInput);
+  editInput.focus();
```

Comment:
[OPTIONAL] Currently, when your tasks are edited, the changes do not get saved in local storage😉. Kindly do well to fix it as this is meant to be a requirement for this milestone and it makes the user experience of the application even better😉.